appservice/src/github/GitHubRepoListStep.ts

```diff
@@ -7,18 +7,25 @@ import { AzureWizardPromptStep, IAzureQuickPickItem } from 'vscode-azureextensio
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { nonNullProp } from '../utils/nonNull';
-import { createRequestOptions, getGitHubQuickPicksWithLoadMore, gitHubRepoData, gitHubWebResource, ICachedQuickPicks } from './connectToGitHub';
+import { getGitHubQuickPicksWithLoadMore, gitHubRepoData, ICachedQuickPicks } from './connectToGitHub';
 import { IConnectToGitHubWizardContext } from './IConnectToGitHubWizardContext';
 
 export class GitHubRepoListStep extends AzureWizardPromptStep<IConnectToGitHubWizardContext> {
     public async prompt(context: IConnectToGitHubWizardContext): Promise<void> {
         const placeHolder: string = localize('chooseRepo', 'Choose repository');
-        let repoData: gitHubRepoData | undefined;
+        let repoData: gitHubRepoData | string;
         const picksCache: ICachedQuickPicks<gitHubRepoData> = { picks: [] };
-        const requestOptions: gitHubWebResource = await createRequestOptions(context, nonNullProp(context, 'orgData').repos_url);
-        do {
-            repoData = (await ext.ui.showQuickPick(this.getRepositories(context, picksCache, requestOptions), { placeHolder })).data;
-        } while (!repoData);
+
+        let url: string = nonNullProp(context, 'orgData').repos_url;
+        // tslint:disable-next-line: no-constant-condition
+        while (true) {
```

Comment:
Pretty clever, I was wondering why you let repoData be a string too.

Reply:
Yeah the old `createRequestOptions` didn't work with the new types for various reasons, so I cleaned it up a bit. This stuff should probably move to octokit anyways 🤷‍♂️

Reply:
Yeah, was thinking we should move it all to appservice and just have Static Web Apps consume from there.